tasks/go_deps.py

```diff
@@ -185,3 +186,95 @@ def show(ctx: Context, build: str, flavor: str = AgentFlavor.base.name, os: str
 
     for dep in deps:
         print(dep)
+
+
+@task(
+    help={
+        'build': f'The agent build to use, one of {", ".join(BINARIES.keys())}',
+        'flavor': f'The agent flavor to use, one of {", ".join(AgentFlavor.__members__.keys())}. Defaults to base',
+        'os': f'The OS to use, one of {", ".join(GOOS_MAPPING.keys())}. Defaults to host platform',
+        'arch': f'The architecture to use, one of {", ".join(GOARCH_MAPPING.keys())}. Defaults to host architecture',
+        'entrypoint': 'A Go package path, defaults to the entrypoint of the build.',
+        'target': 'A Go package path. If specified, generate a graph from the entrypoint to the target.',
+        'std': 'Whether to include the standard library in the graph',
+        'cluster': 'Whether to group packages by cluster.',
+        'fmt': 'The format of the generated image. Must be accepted by "dot -T". Defaults to "svg".',
+        'auto_open': 'Whether to open the generated graph automatically.',
+    }
+)
+def graph(
+    ctx: Context,
+    build: str,
+    flavor: str = AgentFlavor.base.name,
+    os: str | None = None,
+    arch: str | None = None,
+    entrypoint: str | None = None,
+    target: str | None = None,
+    std: bool = False,
+    cluster: bool = False,
+    fmt: str = "svg",
+    auto_open: bool = True,
+):
+    """
+    Generate a dependency graph of the given build.
+    Requires https://github.com/loov/goda and Graphviz's `dot` tools to be in the PATH.
```

Comment:
What do you think about checking if `goda` / `dot` are in the `PATH` and provides a link if needed? Alternatively, you can install these tools in the invoke command `inv setup`.

Reply:
Yes I agree having a clear error message would be much better, good idea !
I think most people will never use this task (it's very niche), it's probably not useful to install it as part of `inv setup`.